src/wallet/coinselection.cpp

```diff
@@ -395,3 +387,30 @@ CAmount GetSelectionWaste(const std::set<CInputCoin>& inputs, CAmount change_cos
 
     return waste;
 }
+
+CAmount SelectionResult::GetSelectedValue() const
+{
+    CAmount ret = 0;
+    for (const auto& coin : m_selected_inputs) {
+        ret += coin.txout.nValue;
+    }
+    return ret;
+}
+
+void SelectionResult::Clear()
```

Comment:
I don't think this function is ever used. I think I'd prefer to remove it, in general it seems unnecessary to have a `Clear` function instead of replacing the structure with a new, empty one.

Reply:
I had intended to use it in the tests, and with the changes to make `m_selected_inputs` private, it is now being used by those tests.